src/ContentTypeFilterListener.php

```diff
@@ -68,6 +68,12 @@ public function onRoute(MvcEvent $e)
         }
 
         $requestBody = $request->getContent();
+
+        // If the $requestBody happens to be an object, unwrap its actual contents
+        if (is_object($requestBody) && method_exists($requestBody, 'getContents')) {
+            $requestBody = $requestBody->getContents();
```

Comment:
I'd argue instead that we cast the body to a string. This ensures the `empty()` check works, and will work flawlessly with PSR-7 bodies as well (as casting to a string internally calls `getContents()`).

I'll make that change on merge, and add a unit test.